builder/hcloud/step_create_server.go

```diff
@@ -78,6 +78,19 @@ func (s *stepCreateServer) Run(ctx context.Context, state multistep.StateBag) mu
 		Labels:     c.ServerLabels,
 	}
 
+	if c.PublicIPv4 != "" {
+		publicIPv4, _, err := client.PrimaryIP.GetByIP(ctx, c.PublicIPv4)
+		if err != nil {
+			return errorHandler(state, ui, "Could not find PublicIPv4", err)
+		}
```

Comment:
`client.PrimaryIP.GetByIP()` does not return an error if no matching IP is found. The following suggestion handles the error and the case that no ip was found:

```suggestion
		if err != nil {
			return errorHandler(state, ui, "Failed to get specified Public IPv4 address", err)
		}
		if publicIPv4 == nil {
		  return errorHandler(state, ui, "", fmt.Errorf("Could not find PublicIPv4"))
		}
```

Reply:
Thanks. I wrote the error handling code before reading the API docs, thinking that there was a single "get by IP address" call, rather than getting a filtered list of IPs.

Reply:
I have validated the error behaviour now as well.

```
==> hcloud.example: Creating server...
==> hcloud.example: Could not find PublicIPv4
==> hcloud.example: Deleting temporary SSH key...
Build 'hcloud.example' errored after 5 seconds 101 milliseconds: Could not find PublicIPv4
```